janusgraph-test/src/test/java/org/janusgraph/diskstorage/inmemory/ClassicVsCompactComparisonTest.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.inmemory;
+
+import org.janusgraph.core.JanusGraph;
+import org.janusgraph.core.JanusGraphFactory;
+import org.janusgraph.core.JanusGraphTransaction;
+import org.janusgraph.core.JanusGraphVertex;
+import org.janusgraph.core.schema.JanusGraphManagement;
+import org.janusgraph.diskstorage.configuration.ModifiableConfiguration;
+import org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration;
+
+
+/**
+ * This is just a simple test program to illustrate comparative performance and memory footprint (as captured via JMC)
+ * between "classic" InMemory backend and the new improved one
```

Comment:
put the test results here, before and after making the change. maybe this class should be a different commit and should precede this PR

Reply:
Thinking about it, it probably doesn't belong in the actual JanusGraph code at all, as it is not a test we want to run as part of our build etc - it is just a program that illustrates the difference between "old" inmemory backend and "new" one. 
I guess I will add it as a code snippet or a mini-project in my private repo and link here?